class ErrorHandler {
  constructor(logger) {
    this.logger = logger;
  }

  handleError(err, req, res, next) {
    this.logger.error(`Error: ${err.message}`, {
      method: req.method,
      url: req.url,
      error: err,
    });

    if (res.headersSent) {
      return next(err); // Let the default error handler deal with it
    }

    res.status(500).json({
      error: {
        message: 'Internal Server Error',
        code: 'INTERNAL_SERVER_ERROR',
      },
    });
  }

  handleNotFoundError(req, res) {
    res.status(404).json({
      error: {
        message: 'Resource Not Found',
        code: 'RESOURCE_NOT_FOUND',
      },
    });
  }
}

module.exports = ErrorHandler;